draft-irtf-cfrg-hash-to-curve.md

```diff
@@ -1066,47 +1114,35 @@ algorithms.
 
 ## Domain separation requirements {#domain-separation}
 
-When invoking hash\_to\_curve from a higher-level protocol, implementors MUST use domain separation
-({{term-domain-separation}}) to avoid interfering with other protocols
-that also use the hash\_to\_curve functionality.
-Protocols that use encode\_to\_curve SHOULD use domain separation
-if possible, though it is not required in this case.
+When invoking hash\_to\_curve or encode\_to\_curve from a higher-level protocol,
+implementors MUST always use domain separation ({{term-domain-separation}}) to
+avoid interfering with other protocols that also use the same functionality.
 
-Protocols that instantiate multiple, independent random
-oracles based on hash\_to\_curve MUST enforce domain separation between
-those oracles.
-This requirement applies both in the case of multiple oracles to the same curve
-and in the case of multiple oracles to different curves.
-This is because the hash\_to\_base primitive ({{hashtobase}}) requires
-domain separation to guarantee independent outputs.
+Protocols that instantiate multiple, independent hash functions based on
+either hash\_to\_curve or encode\_to\_curve MUST enforce domain separation
+between those hash functions.
+This requirement applies both in the case of multiple hashes to the same
+curve and in the case of multiple hashes to different curves.
+(This is because the hash\_to\_base primitive ({{hashtobase}}) requires
+domain separation to guarantee independent outputs.)
 
 Care is required when choosing a domain separation tag.
-Implementors SHOULD observe the following guidelines:
-
-1. Tags should be prepended to the value being hashed, as in the example
-   in {{term-domain-separation}}.
+Implementors MUST observe the following guidelines:
 
-2. Tags should have fixed length, or should be encoded in a way that makes
-   the length of a given tag unambiguous.
-   If a variable-length tag is used, it should be prefixed with a
-   fixed-length field that encodes the length of the tag.
+1. Tags must be supplied as the DST parameter to hash\_to\_base, as
+   described in {{hashtobase}}.
 
-3. Tags should begin with a fixed protocol identification string.
-   Ideally, this identification string should be unique to the protocol.
+2. Tags should begin with a fixed protocol identification string.
+   This identification string should be unique to the protocol.
 
-4. Tags should include a protocol version number.
+3. Tags should include a protocol version number.
 
-5. For protocols that support multiple ciphersuites, tags should include
+4. For protocols that support multiple ciphersuites, tags must include
```

Comment:
Perhaps we should drop a reference to the ciphersuite section here? (One might read this and think of ciphersuites as part of the protocol in which h2c is used, whereas here we mean a single protocol using multiple h2c suites.)

Reply:
Oh! Good point---but actually probably we should mean both.

Reply:
Clearer now?

Reply:
Yep -- looks good!